src/NBitcoin/ConsensusFactory.cs

```diff
@@ -39,6 +39,13 @@ public class ConsensusFactory
         /// </summary>
         private readonly TypeInfo transactionType = typeof(Transaction).GetTypeInfo();
 
+        private readonly Network network;
```

Comment:
In this case, please remove this field.
`public Consensus Consensus { get; set; }`

Reply:
`Consensus` still gets used in the `Coins` class to check the `IsProofOfStake` property. Maybe we can look at removing this or find a work around in the next PR?